pkg/registry/generic/etcd/etcd.go

```diff
@@ -403,6 +403,7 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 			if lastGraceful > 0 {
 				return out, nil
 			}
+			return out, nil
 			// fall through and delete immediately
```

Comment:
Oh, the addition at line 406 is unintended. I'll remove it (and keep the if block).